src/main/groovy/com/github/hauner/openapi/spring/writer/MethodWriter.groovy

```diff
@@ -100,17 +105,27 @@ class MethodWriter {
     private String createParameters (Endpoint endpoint) {
         def ps = endpoint.parameters.collect {
 
+            def methodDefinition = ''
+
             if (it.withAnnotation ()) {
-                "${createParameterAnnotation (it)} ${it.dataType.name} ${Identifier.toCamelCase (it.name)}"
-            } else {
-                "${it.dataType.name} ${Identifier.toCamelCase (it.name)}"
+                if(apiOptions.beanValidation){
+                    methodDefinition += " ${beanValidationWriter.createAnnotations(it.dataType)} "
+                }
+
+                methodDefinition += " ${createParameterAnnotation (it)} "
```

Comment:
the `it.withAnnotation()` is a question about adding the `@RequestParam` annotation or not. I think the bean validation check should not be nested inside that.